src/Listener/ViewListener.php

```diff
@@ -397,4 +440,70 @@ protected function _deriveFieldFromContext($field)
 
         return $value;
     }
+
+    protected function _getTables()
+    {
+        $action = $this->_action();
+        $tables = $action->config('scaffold.tables');
+        if (empty($tables)) {
+            $connection = ConnectionManager::get('default');
+            $schema = $connection->schemaCollection();
+            $tables = $schema->listTables();
+            ksort($tables);
+
+            $blacklist = $action->config('scaffold.tables_blacklist');
+            if (!empty($blacklist)) {
+                $tables = array_diff($tables, $blacklist);
+            }
+        }
+
+        $normal = [];
+        foreach ($tables as $table => $config) {
+            if (is_string($config)) {
+                $config = ['table' => $config];
+            }
+
+            if (is_int($table)) {
+                $table = $config['table'];
+            }
+
+            if (empty($config['action'])) {
+                $config['action'] = 'index';
+            }
+
+            if (empty($config['title'])) {
+                $config['title'] = Inflector::humanize($table);
+            }
+
+            if (empty($config['controller'])) {
+                $config['controller'] = Inflector::camelize($table);
+            }
```

Comment:
Would be cleaner as:

    $config += [
        'action' => 'index',
        'title' => Inflector::humanize($table),
        'controller' => Inflector::camelize($table)
    ];